### PR TITLE
Keep same QR worker alive as long as possible

### PR DESCRIPTION
**Pull request by jer-sen**
# Why

Today, at each `interval` (default to [16ms ](https://github.com/expo/expo/blob/master/packages/expo-camera/src/useWebQRScanner.ts#L94)!) in order to scan the current image for a code, a request is sent to 'https://cdn.jsdelivr.net/npm/jsqr@1.2.0/dist/jsQR.min.js' to get some source code which is then loaded and executed in a new worker to look for a potential code. This could be optimized to lower network use, memory use and processing.

# How

Set `autoTerminate ` to `false` so that when a scan is done, the worker is kept awake and can be reused for the next scan. Remove `timeout` which is not (only) a timeout for one scan but for keeping the worker awake. Note that, with a timeout, when it's reached during a scan, the scan never ends so no other scan is scheduled a codes are no more detected.

# Test Plan

Before, requests to 'https://cdn.jsdelivr.net/npm/jsqr@1.2.0/dist/jsQR.min.js' every 16ms.
After, just one request when the `Camera` component is loaded and everything works fine (scanning a code, unmounting then remounting the component...).

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-camera/src/useWebQRScanner.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ const qrWorkerMethod = ({ data, width, height }: ImageData): any => {
 function useRemoteJsQR() {
   return useWorker(qrWorkerMethod, {
     remoteDependencies: ['https://cdn.jsdelivr.net/npm/jsqr@1.2.0/dist/jsQR.min.js'],
-    timeout: 5000,
+    autoTerminate: false,
```
Comment (EvanBacon) [inline packages/expo-camera/src/useWebQRScanner.ts:42]: Have you tested on an Android web browser and iOS web browser, the functionality is a bit different between the two which is why we implemented this heavy stuff.

Comment (jer-sen) [reply packages/expo-camera/src/useWebQRScanner.ts:42]: Tested on:

- Safary iOS 15.1
- Firefox Android 11
- Firefox Windows 10
- Chrome Windows 10

Everything seems ok.

What's the difference you are talking about?

**Review by EvanBacon — APPROVED** [review]
LGTM just update the CHANGELOG, and run `yarn prepare` in the expo-camera dir because we commit the build files.

**Review by bbarthec — COMMENTED** [review]
I'm fine with that 👍 
The last part missing is to follow our [contribution guidelines (CHANGELOG entry, rebuilding `build` directory)](https://github.com/expo/expo/blob/master/CONTRIBUTING.md#-before-submitting).

**Comment by jer-sen** [comment]
@EvanBacon @bbarthec is it possible to merge this PR?

**Comment by bbarthec** [comment]
@jer-sen, I would like you to follow our [contribution guidelines](https://github.com/expo/expo/blob/master/CONTRIBUTING.md#-before-submitting) by
- adding missing `CHANGELOG entry`
- and rebuilding `build` directory of the affected package
Then we'll be ready to merge it 👍

**Comment by jer-sen** [comment]
I tried to clone the repo but I get an authentication error (with a sub repository?).
I really don't have time for that, if you are used to push to this repo, it will take you only few minutes, I've done the hardest part with debugging, solving and testing. I let you finished the PR if you want to.

Suggestion for the changelog:
🐛 Bug fixes

    Fix a bug on Web which make the module redownload scripts at https://cdn.jsdelivr.net/npm/jsqr@1.2.0/dist/jsQR.min.js every 16ms as long as the component is rendered

**Comment by bbarthec** [comment]
Superseded by
- https://github.com/expo/expo/pull/17833

**Comment by jer-sen** [comment]
Thanks a lot @bbarthec !

**Pull request closed without merging**